OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/SecurityTokenOperationActivity.java

```diff
@@ -183,6 +185,13 @@ protected void doSecurityTokenInBackground() throws IOException {
 
         switch (mRequiredInput.mType) {
             case SECURITY_TOKEN_DECRYPT: {
+                long tokenKeyId = KeyFormattingUtils.getKeyIdFromFingerprint(
+                        mSecurityTokenHelper.getMasterKeyFingerprint(KeyType.SIGN));
```

Comment:
You are right, I think it would be better to check if  subkey ids match and rename the method